src/quantity.cpp

```diff
@@ -20,8 +21,19 @@ Quantity::Quantity(const MeasurementUnit &converter_) :
 	value(0), unit(converter_.getBaseUnit()), converter(converter_) {}
 
 void Quantity::fromString(const std::string &str) {
-	std::istringstream ss(str);
-	ss>>this->value>>this->unit;
+
+    std::string regexpstr = "([0-9]+)(\\.[0-9]*)? *([^ \t]*)";
+    std::regex reg(regexpstr);
+    std::smatch match;
+    std::regex_search(str, match, reg);
+
+    if ( !match.empty() ) {
```

Comment:
Minor nit: in the rest of the code, there are not spaces between parentheses and conditions (see for example at L38 below). I would use the same convention for consistency